### PR TITLE
[ios] Keyboard refactoring

### PR DESCRIPTION
**Pull request by kambala-decapitator**
## Description

- splits iOS keyboard class into generic and platform-specific
- changes layout of native views from manual to AutoLayout

## Motivation and Context
Preparation and code reuse for tvOS.

## How Has This Been Tested?
Tested entering text on iPhone and iPad.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/darwin/ios-common/DarwinEmbedKeyboardView.mm [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+- (void)setHidden:(BOOL)hidden
+{
+  dispatch_sync(dispatch_get_main_queue(), ^{
+    m_inputTextField.secureTextEntry = hidden;
+  });
+}
+
+- (void)textChanged:(NSNotification*)aNotification
+{
+    if (m_darwinEmbedKeyboard)
```
Comment (Rechi) [inline xbmc/platform/darwin/ios-common/DarwinEmbedKeyboardView.mm]: > Use spaces as tab policy with an indentation size of 2.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#32-indentation

File: xbmc/platform/darwin/ios-common/DarwinEmbedKeyboardView.mm [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      m_darwinEmbedKeyboard->fireCallback(m_inputTextField.text.UTF8String);
```
Comment (Rechi) [inline xbmc/platform/darwin/ios-common/DarwinEmbedKeyboardView.mm]: > Use spaces as tab policy with an indentation size of 2.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#32-indentation

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

**Comment by fuzzard** [comment]
Jenkins build this please

**Pull request merged**